crates/pixi_trampoline/src/main.rs

```diff
@@ -18,39 +20,40 @@ struct Metadata {
     env: HashMap<String, String>,
 }
 
-fn read_metadata(current_exe: &Path) -> Metadata {
+fn read_metadata(current_exe: &Path) -> miette::Result<Metadata> {
     // the metadata file is next to the current executable parent folder,
     // under trampoline_configuration/current_exe_name.json
-    let exe_parent = current_exe.parent().expect("should have a parent");
-    let exe_name = current_exe.file_stem().expect("should have a file name");
-    let metadata_path = exe_parent.join(TRAMPOLINE_CONFIGURATION).join(format!("{}{}", exe_name.to_string_lossy(), ".json"));
-    let metadata_file = File::open(metadata_path).unwrap();
-    let metadata: Metadata = serde_json::from_reader(metadata_file).unwrap();
-    metadata
+    if let Some(exe_parent) = current_exe.parent(){
+        let metadata_path = exe_parent.join(TRAMPOLINE_CONFIGURATION).join(format!("{}{}", executable_from_path(current_exe), ".json"));
+        let metadata_file = File::open(&metadata_path).into_diagnostic().wrap_err(format!("Couldn't open {:?}", metadata_path))?;
+        let metadata: Metadata = serde_json::from_reader(metadata_file).into_diagnostic()?;
+        return Ok(metadata);
+    }
+    miette::bail!("Couldn't get the parent folder of the current executable: {:?}", current_exe);
 }
 
-fn prepend_path(extra_path: &str) -> String {
-    let path = env::var("PATH").unwrap();
+fn prepend_path(extra_path: &str) -> miette::Result<String> {
+    let path = env::var("PATH").into_diagnostic().wrap_err("Couldn't get 'PATH'")?;
     let mut split_path = env::split_paths(&path).collect::<Vec<_>>();
     split_path.insert(0, PathBuf::from(extra_path));
-    let new_path = env::join_paths(split_path).unwrap();
-    new_path.to_string_lossy().into_owned()
+    let new_path = env::join_paths(split_path).into_diagnostic().wrap_err("Couldn't join PATH's")?;
```

Comment:
maybe we could also format split_path in the error message

Reply:
Done